main.c

```diff
@@ -52,17 +59,31 @@ static void pointer_handle_leave(void *data, struct wl_pointer *wl_pointer,
 	seat->current_output = NULL;
 }
 
+static void seat_set_outputs_dirty(struct slurp_seat *seat) {
+	if (seat->button_state != WL_POINTER_BUTTON_STATE_PRESSED) {
+		return;
+	}
+	struct slurp_box box;
+	seat_get_box(seat, &box);
+	struct slurp_output *output;
+	wl_list_for_each(output, &seat->state->outputs, link) {
+		if (box_intersect(&output->logical_geometry, &box)) {
+			set_output_dirty(output);
+		}
+	}
+}
+
 static void pointer_handle_motion(void *data, struct wl_pointer *wl_pointer,
 		uint32_t time, wl_fixed_t surface_x, wl_fixed_t surface_y) {
 	struct slurp_seat *seat = data;
+	// the places the cursor moved away from are also dirty
```

Comment:
We should probably only do all of this if `seat->button_state == WL_POINTER_BUTTON_STATE_PRESSED`?

Reply:
Storing x and y is important if the button is not pressed.